src/common/components/signUpModal/signUpModal.component.js

```diff
@@ -63,49 +76,75 @@ class SignUpModalController {
           const step = this.currentStep || 1
           // Split the form into multiple steps for better user experience.
           // Retain the values entered by the user in each step.
-          // Pre-populate the form fields with existing user details to save time.
+          // Pre-populate the form fields with existing user details.
+
+          const organizationNameField = this.$scope.accountType === 'organization' ? [{
+            name: 'organizationName',
+            type: 'text',
+            'label-top': true,
+            label: this.organizationNameTxt,
+            required: true,
+            maxLength: 50,
+            value: this.$scope.organizationName ?? donorData['organization-name'] ?? ''
+          }] : [];
+
           const steps = {
-            // Step 1: Name and email
+            // Step 1: Name, email and account type
             1: [
               {
                 ...schema[0],
-                value: this.$scope.firstName ?? donorData.name['given-name'] ?? this.sessionService.session.first_name ?? ''
+                value: this.$scope.firstName ?? donorData.name?.['given-name'] ?? this.sessionService.session.first_name ?? ''
               },
               {
                 ...schema[1],
-                value: this.$scope.lastName ?? donorData.name['family-name'] ?? this.sessionService.session.last_name ?? ''
+                value: this.$scope.lastName ?? donorData.name?.['family-name'] ?? this.sessionService.session.last_name ?? ''
               },
               {
                 ...schema[2],
                 value: this.$scope.email ?? donorData.email ?? this.sessionService.session.email ?? ''
-              }
+              },
+              {
+                name: 'accountType',
+                type: 'select',
+                options: {
+                  household: this.giveAsIndividualTxt,
+                  organization: this.giveAsOrganizationTxt,
+                },
+                'label-top': true,
+                label: 'Account Type',
+                required: true,
+                wide: true,
+                value: this.$scope.accountType ?? 'household',
```

Comment:
Is there a reason why we're not pulling this from `donorData['donor-type']` if it is set there?

Reply:
Good catch